## *********************************************************
## 
## File autogenerated for the super_modified_servo package 
## by the dynamic_reconfigure package.
## Please do not edit.
## 
## ********************************************************/

from dynamic_reconfigure.encoding import extract_params

inf = float('inf')

config_description = {'upper': 'DEFAULT', 'lower': 'groups', 'srcline': 233, 'name': 'Default', 'parent': 0, 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'cstate': 'true', 'parentname': 'Default', 'class': 'DEFAULT', 'field': 'default', 'state': True, 'parentclass': '', 'groups': [], 'parameters': [{'srcline': 259, 'description': 'Motor ID', 'max': 255, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'motor_id', 'edit_method': '', 'default': 5, 'level': 0, 'min': 4, 'type': 'int'}, {'srcline': 259, 'description': 'New motor ID', 'max': 255, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'new_motor_id', 'edit_method': '', 'default': 5, 'level': 0, 'min': 4, 'type': 'int'}, {'srcline': 259, 'description': 'Set baud rate', 'max': 115200, 'cconsttype': 'const int', 'ctype': 'int', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'baudRate', 'edit_method': '', 'default': 115200, 'level': 0, 'min': 600, 'type': 'int'}, {'srcline': 259, 'description': 'P gain', 'max': 600.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'p_gain', 'edit_method': '', 'default': 145.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'I gain', 'max': 600.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'i_gain', 'edit_method': '', 'default': 15.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'D gain', 'max': 600.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'd_gain', 'edit_method': '', 'default': 40.0, 'level': 0, 'min': 0.0, 'type': 'double'}, {'srcline': 259, 'description': 'Velocity set point', 'max': 10.0, 'cconsttype': 'const double', 'ctype': 'double', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'set_point', 'edit_method': '', 'default': 0.0, 'level': 0, 'min': -10.0, 'type': 'double'}, {'srcline': 259, 'description': 'Send command', 'max': '', 'cconsttype': 'const char * const', 'ctype': 'std::string', 'srcfile': '/home/azisi/ros_catkin_ws/install_isolated/lib/python2.7/site-packages/dynamic_reconfigure/parameter_generator.py', 'name': 'command', 'edit_method': '', 'default': '', 'level': 0, 'min': '', 'type': 'str'}], 'type': '', 'id': 0}

min = {}
max = {}
defaults = {}
level = {}
type = {}
all_level = 0

#def extract_params(config):
#    params = []
#    params.extend(config['parameters'])    
#    for group in config['groups']:
#        params.extend(extract_params(group))
#    return params

for param in extract_params(config_description):
    min[param['name']] = param['min']
    max[param['name']] = param['max']
    defaults[param['name']] = param['default']
    level[param['name']] = param['level']
    type[param['name']] = param['type']
    all_level = all_level | param['level']

